packages/core/src/storage/IndyStorageService.ts

```diff
@@ -42,9 +42,16 @@ export class IndyStorageService<T extends BaseRecord> implements StorageService<
         } else {
           transformedTags[tagName] = [tagValue]
         }
-      } else if (value === '1' || value === '0') {
+      }
+      // Transform '1' and '0' to boolean
+      else if (value === '1' || value === '0') {
         transformedTags[key] = value === '1'
       }
+      // If 1 or 0 is prefixed with 'n__' we need to remove it. This is to prevent
+      // casting the value to a boolean
+      else if (value === 'n__1' || value === 'n__0') {
+        transformedTags[key] = value === 'n__1' ? '1' : '0'
+      }
```

Comment:
Hmm not so sure on the benefit of this. As you say it will be less safe, with no clear advantages.

I'm going to leave it like this for now, if you feel strongly about this point, feel free to convince me

Reply:
I do not feel strongly about this. Just not a big fan of these switch cases. It could still be a ternary expressions, but I don't think that that will improve readability.